core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/wdlparsing/NonCachingWDLParser.scala

```diff
@@ -0,0 +1,18 @@
+package org.broadinstitute.dsde.rawls.jobexec.wdlparsing
+
+import cromwell.client.model.WorkflowDescription
+import org.broadinstitute.dsde.rawls.config.WDLParserConfig
+import org.broadinstitute.dsde.rawls.dataaccess.CromwellSwaggerClient
+import org.broadinstitute.dsde.rawls.model.UserInfo
+
+import scala.concurrent.ExecutionContext
+import scala.util.Try
+
+class NonCachingWDLParser(wdlParsingConfig: WDLParserConfig, cromwellSwaggerClient: CromwellSwaggerClient) extends WDLParser {
+
+  def parse(userInfo: UserInfo, wdl: String)(implicit executionContext: ExecutionContext): Try[WorkflowDescription] = {
+    Try { cromwellSwaggerClient.describe(userInfo, wdl) }
+  }
+
+
+}
```

Comment:
could/should this class disappear, and instead have `CachingWDLParser` accept a constructor argument of `useCache: Boolean`? (and potentially rename "CachingWDLParser")

If you end up adding custom metrics reporting, retry logic, and/or logging, you'd have to copy-paste it to both classes.

Reply:
I think there are trade-offs to having 2 classes for caching and non-caching or 1 class with a useCache argument.

If we have 2 classes, we keep the code for the caching and non-caching pathways separate, which I thing would help for debugging. It also means we're not evaluating a `useCache` argument for each call. We also avoid more code complexity around whether or not we should instantiate the cache. 

If we have both caching and noncaching in the same class, we have fewer files and metrics, retries and logging are in the same place. 

However, I think I can move the retrying to the CromwellSwaggerClient and remove it from here. And I think, since the noncaching parser is just calling the client, it would have minimal logging and metrics anyway.

So, in conclusion, my instinct is to keep it as 2 different classes.

Reply:
works for me - I don't feel super strong about this